siliconcompiler/core.py

```diff
@@ -4996,22 +4996,33 @@ def _record_metric(self, step, index, metric, value, source, source_unit=None):
             self.add('tool', tool, 'task', task, 'report', metric, source, step=step, index=index)
 
     #######################################
-    def _clear_metric(self, step, index, metric):
+    def _clear_metric(self, step, index, metric, preserve=[]):
```

Comment:
preserve needs to be None, putting mutable objects in the defaults for keywords is discouraged.